### PR TITLE
os/bluestore: simplify the logic to loop a collection 

### PR DESCRIPTION
**Pull request by liupan1111**
Signed-off-by: Pan Liu <pan.liu@istuary.com>

### CODE REVIEW

**Comment by liupan1111** [comment]
@tchaikov, please help take a look. Thanks.

File: src/os/bluestore/BlueStore.cc [hunk 1]
```diff
@@ -6941,11 +6941,11 @@ void BlueStore::_txc_add_transaction(TransContext *txc, Transaction *t)
...
   _dump_transaction(t);
 
-  vector<CollectionRef> cvec(i.colls.size());
+  size_t coll_size = i.colls.size();
+  vector<CollectionRef> cvec(coll_size);
   unsigned j = 0;
-  for (vector<coll_t>::iterator p = i.colls.begin(); p != i.colls.end();
-       ++p, ++j) {
-    cvec[j] = _get_collection(*p);
+  for (; j < coll_size; j++) {
```
Comment (tchaikov) [inline src/os/bluestore/BlueStore.cc]: i don't see why this change improves the readability.

Comment (liupan1111) [inline src/os/bluestore/BlueStore.cc]: er... I think only one variable changed with "for" loop, doesn't need p and j.

If you prefer current style, I am fine.
